src/vs/workbench/services/runtimeSession/common/runtimeSession.ts

```diff
@@ -309,7 +309,8 @@ export class RuntimeSessionService extends Disposable implements IRuntimeSession
 			.find(({ info }) =>
 				info.session.runtimeMetadata.runtimeId === runtimeId &&
 				info.session.metadata.sessionMode === LanguageRuntimeSessionMode.Console &&
-				(includeExited || info.state !== RuntimeState.Exited)
+				(includeExited || info.state !== RuntimeState.Exited) &&
+				info.state !== RuntimeState.Uninitialized
```

Comment:
The test case where `session.start()` errors leaves the session in an uninitialized state, which caused `selectRuntime` to return the uninitialized session instead of trying to start a new one. I'm not sure if making the change here will have unintended consequences. We could also update `selectRuntime` to exclude existing uninitialized sessions.